### ISSUE TITLE
Add related projects/dependents/dependency graph

### ISSUE THREAD

**Issue by Lykrast** [modrinth/code#2404]
### Please confirm the following.

- [X] I checked the [existing issues](https://github.com/modrinth/code/issues) for duplicate feature requests
- [X] I have checked that this feature request is not on our [roadmap](https://roadmap.modrinth.com)

### What parts of Modrinth is your feature request related too?

Website

### Is your suggested feature related to a problem? Please describe.

When going to a mod version's download, you can see dependencies for that version (well if the author filled them).

I would like something like curseforge's "dependants" page, where it shows mods/projects that are dependant on the mod you're checking.

I see two uses case for it:

* Looking for addons for a mod, for example going to Farmer's Delight it should show me Crabber's Delight, Ube's Delight, Brazilian Delight and more *(I checked for projects that did fill the dependencies properly)*
* Looking for modpacks containing a specific mod, for example if I go on YUNG's Bridges page it should show me Reminiscent Create *(I'd personally mostly use it to check if anyone puts my mods in modpacks cause am curious)*

### Describe the solution you'd like

While I'd love just a magic button "show me mods that depends on this mod" or "show me modpacks containing this mod" button, I know curseforge's dependant pages has some flaws.

My main frustrations with curseforge's one:

* Can't sort the dependant projects (except alphabetically), often I wished I could sort it by downloads (so like "show me most popular modpack with this mod")
* Can't sort by loader/version (like "show me 1.19.2 (neo)forge modpacks with this mod")

Which now makes it sounds like it could use the search UI here (with the added mess that lots of author don't/won't fill dependant pages properly, though for modpack if it's automatic I guess that part should work?).

### Describe alternatives you've considered

I dunno but writing it down properly it sounds like something that will require a lot of designing, so if you do make that I'm sending you love and appreciation.

### Additional context

I tried my best to see if anyone already suggested that but honestly I have no clue how to properly search for it.

**Comment by maDU59** [comment]
+1 
I'm not creating a new issue to avoid duplicates, but this would be really cool.

**Comment by axolotl-git** [comment]
I agree! It would be a very neat feature, +1

**Comment by maDU59** [comment]
There is no official button on Modrinth, but this website seems to do the same: https://modrinth-statistics.tobinio.dev/reverseSearch

**Comment by randomusert** [comment]
yeah. i really would love to see this as a creator for both Modrinth and CurseForge

**Comment by iHoriZonT** [comment]
I couldn't find this suggestion sorry for duplicate. But man, 2 years?
